### PR TITLE
commit problems + test 52 to 71

### PR DESCRIPTION
**Pull request by IvanMenov**

### CODE REVIEW

**Comment by javadev** [comment]
There are some code smells were found.

https://sonarcloud.io/project/issues?id=javadev_LeetCode-in-Java&pullRequest=27&resolved=false&types=CODE_SMELL

File: src/main/java/s0053_maximum_subarray/Solution.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package s0053_maximum_subarray;
+
+public class Solution {
+    public int maxSubArray(int[] nums) {
+        int maxi = Integer.MIN_VALUE;
+        int sum = 0;
+        for (int i = 0; i < nums.length; i++) {
+            sum += nums[i]; // calculating sub-array sum
+            maxi = Math.max(sum, maxi);
+            if (sum < 0) { // there is no point to carry a -ve subarray sum. hence setting to 0
```
Comment (javadev) [inline src/main/java/s0053_maximum_subarray/Solution.java:10]: Please copy the comment above.

File: src/main/java/s0053_maximum_subarray/Solution.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package s0053_maximum_subarray;
+
+public class Solution {
+    public int maxSubArray(int[] nums) {
+        int maxi = Integer.MIN_VALUE;
+        int sum = 0;
+        for (int i = 0; i < nums.length; i++) {
+            sum += nums[i]; // calculating sub-array sum
```
Comment (javadev) [inline src/main/java/s0053_maximum_subarray/Solution.java:8]: Please copy the comment above.

File: src/main/java/s0054_spiral_matrix/Solution.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        int R = matrix.length - 1;
+        int C = matrix[0].length - 1;
+        while (r <= R && c <= C) {
+            for (int i = c; i <= C; i++) list.add(matrix[r][i]);
+            r++;
+            for (int i = r; i <= R; i++) list.add(matrix[i][C]);
+            C--;
+            for (int i = C; i >= c && r <= R; i--) list.add(matrix[R][i]);
+            R--;
+            for (int i = R; i >= r && c <= C; i--) list.add(matrix[i][c]);
```
Comment (javadev) [inline src/main/java/s0054_spiral_matrix/Solution.java:20]: Please add brackets to the for().

File: src/main/java/s0055_jump_game/Solution.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        int sz = nums.length;
+        int tmp = 1; // we set 1 so it won't break on the first iteration
+        for (int i = 0; i < sz; i++) {
+            tmp--; // we always deduct tmp for every iteration
+            if (tmp < 0) {
+                return false; // if from previous iteration tmp is already 0, it will be <0 here
+                // leading to false value
+            }
+
+            tmp = Math.max(tmp, nums[i]); // we get the maximum value because this value is supposed
```
Comment (javadev) [inline src/main/java/s0055_jump_game/Solution.java:14]: Please move the comment above.

File: src/main/java/s0056_merge_intervals/Solution.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                new Comparator<int[]>() {
+                    public int compare(int a[], int b[]) {
+                        if (a[0] == b[0]) return Integer.compare(a[1], b[1]);
+                        return Integer.compare(a[0], b[0]);
+                    }
+                });
+        /*
+        for(int i=0;i<intervals.length;i++)
+            System.out.println(intervals[i][0]+" "+intervals[i][1]);
+        */
```
Comment (javadev) [inline src/main/java/s0056_merge_intervals/Solution.java:23]: Please remove commented code.

File: src/main/java/s0056_merge_intervals/Solution.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        // System.out.println(list);
+        int arr[][] = new int[list.size()][2];
+        i = 0;
+        for (List<Integer> l : list) {
+            arr[i][0] = l.get(0);
+            arr[i][1] = l.get(1);
+            i++;
+        }
+        return arr;
+        // if you find this helpful,please up vote.
```
Comment (javadev) [inline src/main/java/s0056_merge_intervals/Solution.java:50]: It may be removed.

File: src/main/java/s0057_insert_interval/Solution.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package s0057_insert_interval;
+
+public class Solution {
+    public int[][] insert(int[][] intervals, int[] newInterval) {
+        int n = intervals.length;
+        int l = 0;
+        int r = n - 1;
+        while (l < n && newInterval[0] > intervals[l][1]) l++;
+        while (r >= 0 && newInterval[1] < intervals[r][0]) r--;
```
Comment (javadev) [inline src/main/java/s0057_insert_interval/Solution.java:9]: Please add brackets for while() {}

File: src/test/java/s0054_spiral_matrix/SolutionTest.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.Arrays;
+import org.junit.Test;
+
+public class SolutionTest {
+    @Test
+    public void spiralOrder() {
+        int[][] input = new int[][] {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
```
Comment (javadev) [inline src/test/java/s0054_spiral_matrix/SolutionTest.java:12]: It may be ` int[][] input = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};`

**Review by javadev — CHANGES_REQUESTED** [review]

**Comment by javadev** [comment]
Only one code smell was found.

image

File: src/main/java/s0061_rotate_list/Solution.java [hunk 9]
```diff
@@ -23,10 +23,14 @@ public ListNode rotateRight(ListNode head, int k) {
         for (int i = 1; i <= count - times - 1 && temp != null; i++) {
             temp = temp.next;
         }
-
-        ListNode newHead = temp != null ? temp.next : null;
-        temp.next = null;
-        tail.next = head;
+        ListNode newHead = null;
+        if(temp != null && tail != null) {
```
Comment (javadev) [inline src/main/java/s0061_rotate_list/Solution.java:27]: Please reformat source codes.

**Comment by IvanMenov** [comment]
> Only one code smell was found.
> 
> image

**Pull request merged**